api/swagger/forms.yaml

```diff
@@ -137,6 +203,154 @@ paths:
           $ref: '#/components/responses/InternalServer'
         default:
           "$ref": "#/components/responses/default"
+  /submit/status/{id}:
+    get:
+      operationId: trackFormSubmission
+      tags:
+        - Track Submit Status
+      parameters:
+        - "$ref": "#/components/parameters/formId"
+        - "$ref": "#/components/parameters/submissionId"
+      responses:
+        '200':
+          description: Submission Status
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/SubmitResult'
+                  - type: object
+                    properties:
+                      submitStatus:
+                        type: string
+                        enum:
+                          - PENDING
+                          - SUCCESSFUL
+                          - FAILURE
+                          - EXPIRED
+                      signInfo:
+                        description: It contains information about signing process. It can eb further used to track signing status from Adobe Sign
+                        type: object
+                        properties:
+                          agreementId:
+                            type: string
+                      validationErrors:
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/ValidationError'
+                      metadata:
+                        description: This can contain any additional properties which custom post submit action invocation returns that can be further used to the post submit action status.
+                        type: object
+                        additionalProperties:
+                          type: string
+        '400':
+          $ref: '#/components/responses/GenericBadRequest'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '408':
+          $ref: '#/components/responses/RequestTimeout'
+        '500':
+          $ref: '#/components/responses/InternalServer'
+  /validate/{id}:
```

Comment:
@rismehta  Correct me if I'm wrong but I believe this is not yet implemented. We should probably remove this section.